lib/api/apiUtils/authorization/aclChecks.js

```diff
@@ -86,11 +86,58 @@ function isBucketAuthorized(bucket, requestType, canonicalID) {
     requestType === 'objectGet' || requestType === 'objectHead');
 }
 
+function _isPermissionGranted(permissionList, canonicalID) {
+    if (permissionList.indexOf(publicId) > -1) {
+        return true;
+    }
+
+    if (permissionList.indexOf(allAuthedUsersId) > -1 &&
+        canonicalID !== publicId) {
+        return true;
+    }
+
+    if (permissionList.indexOf(canonicalID) > -1) {
+        return true;
+    }
+
+    return false;
+}
+
+function hasBucketReadAccess(bucket, requestType, canonicalID) {
+    const bucketAcl = bucket.getAcl();
+    const bucketOwner = bucket.getOwner();
+
+    // User is already authorized on the bucket for FULL_CONTROL or WRITE or
+    // bucket has canned ACL public-read-write
+    if (requestType === 'objectPut' || requestType === 'objectDelete') {
```

Comment:
> I find it confusing to have this check done in this helper function, i.e. checking write requests in a "has read access?" checker

I can see how that can be confusing; the idea is that if this wasn't blocked at the bucket level, the user should have read access, but this would not be true if the user was only granted `WRITE` permissions

Reply:
moved `objectPut` and `objectDelete` check out of the `read` check;